Maint/be loud about key usage

gchronis opened the pull request:
Prior to this PR, the secure git commands required use of the user's GPG signing key many times to sign RSL entries, stash work, and sign commits. It gave no visible clue as to why it needed the the signatures. This PR adds explanations to stdout which describe the process of the tools, including statements of impending GPG key requests along with what the key will be used to do. It also decreases the number of signatures necessary by temporarily turning off gpg signing during the 'stash local work phase', as there is no need to sign the short-lived stash that is created only for the duration of command execution.

Closes Issue #36

Commit:
Alert user to GPG key usage with explanation

Commit:
Don't GPG sign stash commits

Commit:
Be specific about # of signatures expected

Prior to this commit, merge and promote used secure-fetch and secure-push
without giving any insight to their internal workings. This commit prints the
number of signatures that will be required for their use.

jeffweiss commented on git-merge-pr:
```diff
@@ -100,9 +100,10 @@ if [ $CURRENT_BRANCH = $BRANCH_TO_MERGE ] &&  [ $PRUNE_FLAG = off ]; then
 
 echo "Stashing any local changes and checking out remote branch 'origin/$DESTINATION_BRANCH'"
-STASH_OUTPUT="$(git stash save -a)" || exit 1;
+STASH_OUTPUT="$(git -c commit.gpgSign=false stash save -a)" || exit 1;
```
what?!? heresy!

gchronis replied:
is this bad? why would we need to gpgsign stash commits?

gchronis replied:
update from Jeff: this is not bad

jeffweiss replied:
Not bad

jeffweiss commented on git-merge-pr:
```diff
@@ -100,9 +100,10 @@ if [ $CURRENT_BRANCH = $BRANCH_TO_MERGE ] &&  [ $PRUNE_FLAG = off ]; then
+STASH_OUTPUT="$(git -c commit.gpgSign=false stash save -a)" || exit 1;
 
 echo "Securely fetching remote branches and revisions"
+echo "GPG signature required to commit RSL fetch entry"
```
Do we want to preface this somehow so it's offset from otherwise normal progress?
`<--- GPG signature required...`
`\o/ GPG signature required...`
`<-----> GPG signature required...`

gchronis replied:
Sounds like we do

jeffweiss commented on git-merge-pr:
```diff
@@ -115,6 +116,7 @@ run_command "git merge -S --no-ff -m $MERGE_MESSAGE origin/$BRANCH_TO_MERGE" ||
 echo "Merge Successful!"
 
 echo "Attempting to push to remote repository 'origin'"
+echo "2 GPG signatures required to clearsign RSL push entry and sign RSL commit"
```
Again, we might want to preface so folks know there's an action required... I don't actually expect them to read...

Commit:
Merge remote-tracking branch 'origin/devel' into maint/be-loud-about-key-usage

Commit:
offset GPG signature requirements to grab attention

Commit:
Break up progress statements in git-merge-pr

This commit makes the stdout progress statements easier to parse, so that the
user is mor elikely to notice when action is required. Also, this commit fixes a
in incorrect variable name used to fetch the feature branch to be merged.

Commit:
Break up progress statements in git-promote

This commit makes progress messages easier to parse, in order to make users
more likely to notice when action is required. This commit also fetches the
latest revision of the target branch before attempting to ff changes over it.
